Lesson24-signup-form/assets/assets/feyza/assets/app.js

```diff
@@ -0,0 +1,35 @@
+const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+const form = document.getElementById("form");
+
+form.addEventListener("submit", function (e) {
+  e.preventDefault();
+
+  let isEmailCorrect = true;
+  const emailInput = document.getElementById("email");
+  const emailError = document.getElementById("emailError");
+  const formContainer = document.querySelector("box");
+  const hidenBox = document.getElementsByClassName("hide-box");
```

Comment:
Here, a collection of elements is assigned to the 'hidenBox' variable so the code does not work. When you have a collection of elements, you'd need to use a loop or `map` function to iterate through each of them.

However, in this case, you only have one element and you only expect to select one element. Therefore, using a selector function that selects multiple elements is not necessary. Add id to the HTML element (better) or replace 'getElementsByClassName' with 'querySelector' https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector


If you use `querySelector`, you'd need to change the code like this:
`const hidenBox = document.querySelector(".hide-box");`